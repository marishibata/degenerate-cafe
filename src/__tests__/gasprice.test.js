import React from 'react';
import { render } from '@testing-library/react';
import GasPrice from '../Components/gasprice';
import WS from 'jest-websocket-mock'


let ws;
beforeEach(() => {
  ws = new WS('wss://www.gasnow.org/ws/gasprice');
});
afterEach(() => {
  WS.clean();
});


describe('The Gas Price component', () => {
  it('renders loading before connection', async () => {
    const { findByText, debug } = render(<GasPrice/>);
    const result = await findByText('loading');
    expect(result).toBeTruthy();
    await ws.connected;
    debug();
  })
})