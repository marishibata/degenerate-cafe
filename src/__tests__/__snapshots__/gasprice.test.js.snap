// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Gas Price component Renders consistently 1`] = `
<DocumentFragment>
  <div
    class="par-gas"
  >
    <p>
      loading
    </p>
     
  </div>
</DocumentFragment>
`;
